.github/workflows/package-release.yml

```diff
@@ -0,0 +1,64 @@
+name: sgc-release
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  sgcreleaser:
+    name: secretgen controller release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.6
+
+      - name: Install Carvel Tools
+        uses: vmware-tanzu/carvel-setup-action@v1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Setting up minikube
+        run: |
+          set -e -x
+          minikube start --driver=docker --wait=all
+          docker buildx create minikube --use --driver=kubernetes --bootstrap
+
+      - name: Fetch kctrl
+        uses: actions/checkout@v3
+        with:
+          ref: kctrl-pkg-init-release
+          repository: vmware-tanzu/carvel-kapp-controller
+          path: carvel-kapp-controller
+          fetch-depth: 0
+
+      - name: Build kctrl
+        working-directory: ./carvel-kapp-controller/cli
+        run: |
+          ./hack/build.sh
+
+      - name: Run Package build
+        run: |
+          ./carvel-kapp-controller/cli/kctrl pkg release -y -v ${{ github.ref_name }} --debug
+      
+      - name: Create draft release with package and metadata
+        uses: softprops/action-gh-release@17cd0d34deddf848fc0e7d9be5202c148c270a0a
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          files: |
+            carvel-artifacts/packages/secretgen-controller.carvel.dev/metadata.yml
+            carvel-artifacts/packages/secretgen-controller.carvel.dev/package.yml
+          draft: true
```

Comment:
There are test running on generated PackageCR in `release.yml` which is missing.

Reply:
Modified release.yml itself to generate Package and PackageMetada using kctrl